### PR TITLE
Test the `deserialize` Twig filter

### PR DESCRIPTION
**Pull request by ausi**
Tests for #7139

### CODE REVIEW

**Comment by leofeyer** [comment]
This test is incorrect, which is why I didn‘t add it in the original PR.

The code adds a Twig filter that calls a method of the Contao framework. A test should test exactly that. However, the suggested test does not do this and instead tests 1. that the `render()` method correctly replaces variables and 2. that the `deserialize()` method correctly unserializes the given array.

No. 1 is already covered by the Twig test suite and no. 2 should be tested in the `StringUtilTest` class and not in a Twig related test. And besides this, why do we only test the `deserialize` filter? What about `escape`, `insert_tag`, `highlight`, `format_bytes`, `sanitize_html`, `csp_inline_styles` and `encode_email`?

**Comment by ausi** [comment]
> The code adds a Twig filter that calls a method of the Contao framework. A test should test exactly that. However, the suggested test does not do this and instead tests 1. that the `render()` method correctly replaces variables and 2. that the `deserialize()` method correctly unserializes the given array.

I thought that the `TwigIntegrationTest` is meant to test exactly that - the whole integration - not only the single unit of the Twig part.

> And besides this, why do we only test the `deserialize` filter?

We already test the filters `highlight`, `highlight_auto`, `insert_tag_raw` and `raw|insert_tag` in that same test class so I thought adding the deserialize filter there would make sense.

File: core-bundle/tests/Twig/TwigIntegrationTest.php [hunk 1]
```diff
@@ -310,4 +310,114 @@ public function testPreservesSafetyInInsertTagFilters(): void
...
+            ),
+        );
+
+        $output = $environment->render('test.html.twig', [
+            'values' => $values,
+        ]);
+
+        $this->assertSame($expectedOutput, $output);
+
+        $this->resetStaticProperties([Highlighter::class]);
```
Comment (ausi) [inline core-bundle/tests/Twig/TwigIntegrationTest.php]:
```suggestion
```

**Comment by ausi** [comment]
Also, the main goal of the test for me was also to make sure the Twig integration is correct, meaning it allows `mixed` as input, not only `string` and always returns an `array`.

**Comment by leofeyer** [comment]
> meaning it allows `mixed` as input, not only `string` and always returns an `array`.

That is a valid point. I this case, I agree with the test.

**Review by m-vo — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**